### PR TITLE
Fix element chaining in async mode

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

Fix issue when chaining in some cases return promise.
#4254 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mgrybyk** [comment]
added also smoke test for it

File: packages/wdio-webdriver-mock-service/src/index.js [hunk 1]
```diff
@@ -55,6 +56,15 @@ export default class WebdriverMockService {
         this.command.elementClick(ELEMENT_ID).once().reply(200, { value: null })
     }
 
+    isExistingScenario() {
+        this.nockReset()
+        const elemResponse = { 'element-6066-11e4-a52e-4f735466cecf': ELEMENT_ID }
+
+        this.command.findElement().times(1).reply(200, { value: elemResponse })
+        this.command.findElementFromElement(ELEMENT_ID).times(2).reply(200, { value: elemResponse })
+        this.command.findElementsFromElement(ELEMENT_ID).times(2).reply(200, { value: [elemResponse] })
```
Comment (christian-bromann) [inline packages/wdio-webdriver-mock-service/src/index.js]: this is not needed
```suggestion
```

File: tests/mocha/test.js [hunk 2]
```diff
@@ -21,6 +21,14 @@ describe('Mocha smoke test', () => {
         $('elem').click()
     })
 
+    it('should chain properly', () => {
+        browser.isExistingScenario()
+
+        const el = browser.$('body')
+        el.$('.selector-1').isExisting()
+        el.$('.selector-2').isExisting()
```
Comment (christian-bromann) [inline tests/mocha/test.js]: can we also assert the value to ensure that the promises got resolved?

Comment (mgrybyk) [reply tests/mocha/test.js]: before the fix `el.$('.selector-2')` was Promise, so calling isExisting() was not working.
Anyway it makes sense to add additional verification

File: packages/wdio-webdriver-mock-service/src/index.js (hunk 1 above)
Comment (mgrybyk) [reply packages/wdio-webdriver-mock-service/src/index.js]: can remove this one because calling isExisting() calls findElementsFromElement underneath

Comment (christian-bromann) [reply packages/wdio-webdriver-mock-service/src/index.js]: right

**Pull request merged**